docs/source/integrations/middleware.md

```diff
@@ -36,23 +36,26 @@ const express = require('express');
 const { ApolloServer } = require('apollo-server-express');
 const { typeDefs, resolvers } = require('./schema');
 
-const app = express();
-const server = new ApolloServer({
-  typeDefs,
-  resolvers,
-});
-
-server.applyMiddleware({ app });
-
-app.use((req, res) => {
-  res.status(200);
-  res.send('Hello!');
-  res.end();
-});
-
-app.listen({ port: 4000 }, () =>
-  console.log(`🚀 Server ready at http://localhost:4000${server.graphqlPath}`)
-)
+async function startApolloServer() {
```

Comment:
I mean, the problem is... how are you supposed to call it? The whole point is I don't want to encourage people to call it in a context where errors will turn into `unhandledRejection`. So I could show it being called from top-level await... but then I might as well just revert back to not having a function.  Or I could show it being called with `.catch()` but that seems like it would get distracting (and it's unclear that a particular bit of `.catch()` from our docs is actually the best way to integrate it into your system).

While Node does now support top-level await (as of v14.8.0), it requires you to be in a special "module" mode where your file ends with `.mjs`, or set a flag in package.json. And it's different in TypeScript... here you have to be in module module (which should be the case for any of these examples because of the use of `import`) but you also have to set the `target` and/or `module` tsconfig option appropriately...

It feels like this is a big bundle of worms, compared to "here's an async function and it's your job to know how to use it". Since this mostly affects "advanced" uses (I'm not editing any calls to `apollo-server`'s `listen()`) I'm a little less worried?

I mean, the fact that our `apollo-server` examples tell you to call `listen()` with only `.then` and not `.catch` is another issue, and perhaps we should go through and change those all to use top-level await with explicit guidance to use `.mjs` files, but that's a project for another day...